### PR TITLE
[RJA-1369][feature]: minimum StepCoordinator changes to support creation...

### PR DESCRIPTION
**Pull request by ddurnev**
... of remote ActionExecutor.

Run-local step/action changed as an example of remote action support in plugin.

### CODE REVIEW

File: agent-commons/src/main/resources/application.conf [hunk 1]
```diff
@@ -10,17 +10,16 @@ agent {
...
             #log-received-messages = on
             #log-sent-messages = on
-           netty {
+            netty {
                 #port = 2552
-                #hostname = ""
-                # TODO: provide correct SSL config
-                ssl-key-store = ""
-                ssl-trust-store = ""
+                ssl-key-store = ${user.home}/.keystore
```
Comment (rsvato) [inline agent-commons/src/main/resources/application.conf]: Do we really need it, as well as SSL enabled by default?

File: agent-commons/src/main/scala/com/griddynamics/genesis/workflow/agent/ExecutorActor.scala [hunk 2]
```diff
@@ -48,16 +47,16 @@ class ExecutorActor(unsafeExecutor: AsyncActionExecutor,
     case Start => {
       log.debug("Starting async executor for '%s'", safeExecutor.action)
       safeExecutor.startAsync()
-      cancellable = context.system.scheduler.schedule(0 milliseconds, 2000 milliseconds, self, Beat(Success))
+      cancellable = context.system.scheduler.schedule(0 milliseconds, 2000 milliseconds, self, Beat(Success()))
```
Comment (rsvato) [inline agent-commons/src/main/scala/com/griddynamics/genesis/workflow/agent/ExecutorActor.scala]: We'll need to config it in the future. Why not now?

File: backend/src/main/scala/com/griddynamics/genesis/workflow/actor/StepCoordinator.scala [hunk 3]
```diff
@@ -152,11 +158,44 @@ class StepCoordinator(unsafeStepCoordinator: workflow.StepCoordinator,
...
+  private def executorActor(asyncExecutor: AsyncActionExecutor) = asyncExecutor.action match {
+    case r: RemoteAgentExec if r.tag.nonEmpty => remoteExecutor(r.tag, asyncExecutor.action)
+    case _ =>
+      system.actorOf(Props(new actor.ActionExecutor(asyncExecutor, self, beatSource)))
+   }
+
+  import akka.pattern.ask
+  import akka.util.duration._
+  import akka.dispatch.Await
+  private val TIMEOUT_REMOTE_ACTOR = akka.util.Timeout(10 seconds)
```
Comment (rsvato) [inline backend/src/main/scala/com/griddynamics/genesis/workflow/actor/StepCoordinator.scala]: Again, this val has to be configurable

File: agent-commons/src/main/scala/com/griddynamics/genesis/agent/FrontActor.scala [hunk 4]
```diff
@@ -27,24 +27,30 @@ import akka.actor.{ActorRef, Props, Actor}
...
+        sender ! akka.actor.Status.Failure(t)
+        log.error(t, "Error while processing remote task: %s", rt)
+    }
     case m => log.debug("Unknown message: " + m)
-    // TODO: add signal dispatching
   }
 
-  def startExecutor(executor: ActionExecutor, remote: ActorRef) {
+  def executorActor(executor: ActionExecutor, remote: ActorRef, logger: ActorRef) = {
+    LoggerWrapper.registerRemote(logger)
```
Comment (bugzmanov) [inline agent-commons/src/main/scala/com/griddynamics/genesis/agent/FrontActor.scala]: This doesn't look like a concurrency friendly code. Probably we shouldn't use LoggerWrapper as a singleton anymore. 
And bake some cakes to solve injection problems?

File: agent-commons/src/main/scala/com/griddynamics/genesis/agent/GenesisAgent.scala [hunk 5]
```diff
@@ -34,6 +35,12 @@ class AgentApplication extends akka.kernel.Bootable {
   val log = Logging(system, classOf[AgentApplication])
   val appContext = new ClassPathXmlApplicationContext("classpath:agent-config.xml")
   val frontActor = system.actorOf(Props(appContext.getBean(classOf[AgentContext]).frontActor), "frontActor")
+  val listener = system.actorOf(Props(new Actor {
```
Comment (bugzmanov) [inline agent-commons/src/main/scala/com/griddynamics/genesis/agent/GenesisAgent.scala]: is it neccessary? Akka loggs this events already in debug level.

File: backend/src/main/scala/com/griddynamics/genesis/workflow/actor/StepCoordinator.scala [hunk 6]
```diff
@@ -152,11 +159,48 @@ class StepCoordinator(unsafeStepCoordinator: workflow.StepCoordinator,
...
+    val path = REMOTE_ACTOR_LOOKUP_PATH.format(agent.hostname, agent.port)
+    val remoteFront = system.actorFor(path)
+    implicit val timeout = TIMEOUT_REMOTE_ACTOR
+    val logger = action match {
+      case al: ActionWithLog => al.logger
+      case _ => LoggerWrapper.logger
     }
+    val futureRemote = remoteFront ? RemoteTask(action, self, logger)
+    log.debug("Waiting for remote executor... ")
+    Await.result(futureRemote, timeout.duration).asInstanceOf[ActorRef]
```
Comment (bugzmanov) [inline backend/src/main/scala/com/griddynamics/genesis/workflow/actor/StepCoordinator.scala:191]: side-note: it's more akka-style to use mapTo on futures than using asInstanceOf on results

(remoteFront ? RemoteTask(action, self, logger)).mapTo[ActorRef]

**Pull request merged**